rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -96,6 +96,20 @@ Player::Player(
   storage_options_(storage_options),
   play_options_(play_options)
 {
+  {
+    reader_->open(storage_options_, {"", rmw_get_serialization_format()});
+    double rate = play_options_.rate > 0.0 ? play_options_.rate : 1.0;
```

Comment:
I disagree. `PlayOptions` is passed all the way from the CLI layer, I don't see any reason to force that layer to make multiple calls to configure, when you could configure with a single options struct. Additionally, we want the player to be able to be initialized as a Component Node with Parameters, meaning there is _no way_ to make an additional call in construction, for this configuration.

Reply:
I will also not want to deal with this in this PR - as this line simple refactoring of _existing_ behavior.

Reply:
> start clock with rate equal to 1 on construction since we are not going to run clock right after construction anyway. And then adjust rate only via setter method. 

This part is fine - but outside the scope of this change. PRs welcome

Reply:
Updated to remove rate from the clock constructor and add the check, for now, in the clock implemetnation.